src/main/java/appeng/parts/p2p/PartP2PLight.java

```diff
@@ -81,15 +81,21 @@ public void writeToStream( final ByteBuf data ) throws IOException
 	{
 		super.writeToStream( data );
 		data.writeInt( this.isOutput() ? this.lastValue : 0 );
+		data.writeFloat( this.opacity );
 	}
 
 	@Override
 	public boolean readFromStream( final ByteBuf data ) throws IOException
 	{
 		super.readFromStream( data );
+		final int oldValue = this.lastValue;
+		final float oldOpacity = this.opacity;
+
 		this.lastValue = data.readInt();
+		this.opacity = data.readFloat();
+
 		this.setOutput( this.lastValue > 0 );
-		return false;
+		return lastValue != oldValue || oldOpacity != this.opacity;
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Equality tests should not be made with floating point values. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.orod.org/coding_rules#rule_key=squid%3AS1244)